services/provider/server/server.go

```diff
@@ -633,24 +632,23 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 				return nil, status.Errorf(codes.Internal, "failed to get %s cephFilesystemSubVolumeGroup. %v", cephRes.Name, err)
 			}
 
-			nodeCephClientSecret, _, err := s.getCephClientInformation(ctx, cephRes.CephClients["node"])
-			if err != nil {
-				return nil, status.Error(codes.Internal, err.Error())
-			}
-
-			provisionerCephClientSecret, _, err := s.getCephClientInformation(ctx, cephRes.CephClients["provisioner"])
-			if err != nil {
-				return nil, status.Error(codes.Internal, err.Error())
+			provisionerSecretName := storageClaimCephCsiSecretName("provisioner", storageRequestHash)
+			nodeSecretName := storageClaimCephCsiSecretName("node", storageRequestHash)
+			cephfsStorageClassData := map[string]string{
+				"clusterID":          getSubVolumeGroupClusterID(subVolumeGroup),
+				"subvolumegroupname": subVolumeGroup.Name,
+				"fsName":             subVolumeGroup.Spec.FilesystemName,
+				"pool":               subVolumeGroup.GetLabels()[v1alpha1.CephFileSystemDataPoolLabel],
+				"csi.storage.k8s.io/provisioner-secret-name":       provisionerSecretName,
+				"csi.storage.k8s.io/node-stage-secret-name":        nodeSecretName,
+				"csi.storage.k8s.io/controller-expand-secret-name": provisionerSecretName,
 			}
 
-			storageClassName = "cephfs"
-			storageClassData["clusterID"] = getSubVolumeGroupClusterID(subVolumeGroup)
-			storageClassData["subvolumegroupname"] = subVolumeGroup.Name
-			storageClassData["fsName"] = subVolumeGroup.Spec.FilesystemName
-			storageClassData["pool"] = subVolumeGroup.GetLabels()[v1alpha1.CephFileSystemDataPoolLabel]
-			storageClassData["csi.storage.k8s.io/provisioner-secret-name"] = provisionerCephClientSecret
-			storageClassData["csi.storage.k8s.io/node-stage-secret-name"] = nodeCephClientSecret
-			storageClassData["csi.storage.k8s.io/controller-expand-secret-name"] = provisionerCephClientSecret
+			extR = append(extR, &pb.ExternalResource{
+				Name: "cephfs",
+				Kind: "StorageClass",
+				Data: mustMarshal(cephfsStorageClassData),
+			})
 
 			extR = append(extR, &pb.ExternalResource{
```

Comment:
this section might also be not required, please check it.

Reply:
as discussed offline, this is redundant data being sent which can be removed at a later point.